src/lib/vip-import-validate-files.js

```diff
@@ -142,40 +142,52 @@ const recommendAcceptableFileNames = () => {
 /**
 	* Nested Directory Search
  *
- * Use recursion to identify the nested tree structure of the folders
+ * Use recursion to identify the nested tree structure of the given media file
+	*
+	* Example media file:
+	*  - Given directory: uploads
+	*   - Nested directories: 2020, 2019, 2018, 2017
+	*    - Nested directories: 01, 02, 03, 04, 05, 06
+	*     - Individual files: image.jpg, image2.jpg, etc.
  *
- * @param {string} directory Root directory, or the current directory
+ * @param {string} directory Root directory, or the given (current) directory
  */
-export const findNestedDirectories = async directory => {
-	let dir, nestedDir;
+const files = [];
+const folderStructureObj = {};
+
+export const findNestedDirectories = directory => {
+	let nestedDirectories;
 
 	try {
-		// Read what's inside the current directory
-		dir = await readDir( directory );
+		// Read nested directories within the given directory
+		nestedDirectories = fs.readdirSync( directory );
 
 		// Filter out hidden files such as .DS_Store
-		dir = dir.filter( file => ! ( /(^|\/)\.[^\/\.]/g ).test( file ) );
-
-		const firstNestedFolder = dir[ 0 ]; // The first nested folder of the given directory
-		nestedDir = firstNestedFolder;
-
-		// Once we hit individual media files, stop
-		const regexExtension = /\.\w{3,4}$/;
-		const mediaFiles = regexExtension.test( nestedDir );
-
-		if ( dir !== undefined && mediaFiles ) {
-			return directory;
-		}
+		nestedDirectories = nestedDirectories.filter( file => ! ( /(^|\/)\.[^\/\.]/g ).test( file ) );
+
+		nestedDirectories.forEach( dir => {
+			// Concatenate the file path of the parent directory with the nested directory
+			const filePath = path.join( directory, dir );
+			const statSync = fs.statSync( filePath ); // Get stats on the file/folder
+
+			// Keep looking for nested directories until we hit individual files
+			if ( statSync.isDirectory() ) {
+				findNestedDirectories( filePath );
```

Comment:
Should this have a maximum call depth to pass?
Or can customers have an unlimited number of nested folders inside of the `yyyy/mm/` directory?

Reply:
Somewhat of an edge case but not a bad idea. Maybe we should limit it to "X" and have a warning show if it's reached (and possibly a flag to override that number)

Reply:
I'll do some more research into what the limits are and see if we should add one in a following PR!